tests and message interface

acurrieclark opened the pull request:
Purely to illustrate what I had in mind and to see what you think. I am aware that this is very much a breaking change from your code.

Any feedback would be gratefully received.

acurrieclark committed:
aws client can now be injected

acurrieclark committed:
Added small test suite
Introduced a message object which formats the request properly for android and ios
updated readme

jrhenderson1988 commented:
Hi @acurrieclark - Thank you for the pull request. We really appreciate all the effort you've put into it and we love the direction you've taken. We would like to get this merged into the project, however at the moment we're looking to support PHP 7.0 and above. Some of the code you've submitted looks like it would only be supported by 7.1 and above, particularly the use of nullable types in places. If you would be able to make adjustments to support 7.0 we would definitely look to get your pull request merged in. Thank you for your effort.

acurrieclark commented:
Good point. I will look into that and adjust. Thanks for having a look.

acurrieclark committed:
Add travis config file

acurrieclark committed:
Update .travis.yml

acurrieclark committed:
Update .travis.yml

acurrieclark committed:
Update .travis.yml

acurrieclark committed:
Include composer.lock file

acurrieclark committed:
Updated composer.json to require versions of dependencies which work for multiple versions of php7.x

acurrieclark committed:
Removed need for nullable types in the message class
Updated tests to check zero values still pass

acurrieclark committed:
Added implementation for android 'inbox' mode with tests

acurrieclark committed:
Reverted composer.json to permit testing on PHP 7.0

acurrieclark commented:
I have updated this to reflect requirements for PHP 7.0, although I do not have it running on my local dev setup. The test all pass on Travis CI for both 7.0 and 7.1 , and I have introduced a Travis config file if you wanted to set that up on the root repo too?

As an aside, the message structure which is generated for iOS and Android is the structure which has worked for me in the past and which works well in all my tests. All the apps I have tested with are Cordova base, working with the [Phonegap Push Plugin](https://github.com/phonegap/phonegap-plugin-push). It might be worthwhile, if you have access to one, testing it with a few other apps.

Also, I haven't documented the Message class yet. Will do if I have time, but for now i think this pull request is ready.

Thanks for letting me contribute.

mforcer commented:
Thanks for the update @acurrieclark will review/test asap and get this merged.
